app/src/main/java/com/example/bottomnav/UnsignedLongDecoder.java

```diff
@@ -1,23 +1,17 @@
 package com.example.bottomnav;
 
 import java.math.BigInteger;
-import java.nio.ByteBuffer;
+import java.util.Optional;
 
 public class UnsignedLongDecoder<T> extends IntegerDecoder {
-    Long sign;
+    Long sign = (long) 0xffffffffL;
```

Comment:
Are you sure you need a separate UnsignedLongDecoder? This is a superset of UnsignedIntegerDecoder, but a bit less (inconsequentially) memory efficient since it needs to store a long instead of an int? But the trade-off for this efficiency is that you now have more code the maintain.

In fact, I'd even say that given the expected use case, if we used a BigInteger internally to unify all the int-like decoders that would still be worth the code maintenance tradeoff.